src/plugins/intel_gpu/src/plugin/ops/matmul.cpp

```diff
@@ -161,82 +161,78 @@ static void CreateMatMulOp(Program& p, const std::shared_ptr<ngraph::op::v0::Mat
-                auto reorderPrim = cldnn::reorder(reorderName,
-                                                  inputPrimitives[i],
-                                                  targetFormat,
-                                                  targetDatatype,
-                                                  std::vector<float>(),
-                                                  cldnn::reorder_mean_mode::subtract);
-
-                p.add_primitive(*op, reorderPrim);
-
-                inputPrimitives[i] = reorderName;
-            }
-
-            // Reshape input if they differ or gemm specific shape matches default one
-            if (inputDimsN != outDimsN || inputDimsN < 4) {
-                auto reshapeName = layerName + "_cldnn_in" + std::to_string(i) + "_reshape";
-
-                // Extend input dimensions by prepending ones
-                if (inputDimsN == 1) {
-                    // One-dimensional tensors unsqueezing is applied for each input independently.
-                    // The axes inserted in this step are not included in the output shape.
-                    // * If rank of the **first** input is equal to 1, it is always unsqueezed to 2D tensor **row vector** (regardless of `transpose_a`)
-                    // by adding axes with size 1 at ROW_INDEX_DIM, to the **left** of the shape. For example `[S]` will be reshaped to `[1, S]`.
-                    // * If rank of the **second** input is equal to 1, it is always unsqueezed to 2D tensor **column vector** (regardless of `transpose_b`)
-                    // by adding axes with size 1 at COL_INDEX_DIM, to the **right** of the shape. For example `[S]` will be reshaped to `[S, 1]`.
-                    bool transpose = false;
-                    if (i == 0) {
-                        transpose = op->get_transpose_a();
-                        inputDims.insert(inputDims.begin(), 1);
-                    } else {
-                        transpose = op->get_transpose_b();
-                        inputDims.insert(inputDims.end(), 1);
-                    }
-                    // Specs says that shapes must be unsqueezed regardless of tranpose flag, but primitive implementation always respects transposes
-                    // so we have to swap dimensions correspondingly to have consistent shapes.
-                    if (transpose) {
-                        std::swap(inputDims[0], inputDims[1]);
-                    }
-                }
-                if (inputDimsN < outDimsN)
-                    inputDims.insert(inputDims.begin(), outDimsN - inputDimsN, 1ul);
-
-                auto targetShape = gemmSpecificTensor(inputDims);
-
-                auto reshapePrim = cldnn::reshape(reshapeName, inputPrimitives[i], targetShape);
-
-                p.add_primitive(*op, reshapePrim);
-
-                inputPrimitives[i] = reshapeName;
-            }
-        }
-
+//        auto gemmSpecificTensor = [](const InferenceEngine::SizeVector& dims) {
```

Comment:
??

I assume we need the fix from #13248 instead of that.

Reply:
oh this is missing removal of temporal change I will update it

Reply:
The PR13248 still needs to be fixed for dynamic shape because it cannot get the shape size and there is no need to add reshape/reorder for partial shape usage.. So I applied some fixes.

Reply:
If there are any issues with dynamic gemm in #13248 - let's address them there. Or at least implement your changes on top of that patch or we'll have a conflict which will be not easy to resolve.